### PR TITLE
Data Sync - update rdb query

### PR DESCRIPTION
**Pull request by ndduc01**
## Notes

Please include a summary of the change and which issue is fixed or which feature is added. 

## JIRA

- **Related story**: [Jira Ticket](url)

## Checklist

- [ ] PR focuses on a single story.
- [ ] New unit tests added and ensured they pass.
- [ ] Service has been tested in local and it works as expected.
- [ ] Documentation has been updated for this code change (if needed).
- [ ] Code follows the Java Coding Conventions (https://www.oracle.com/java/technologies/javase/codeconventions-programmingpractices.html).

## Types of changes

What types of changes does this PR introduces?

- [ ] Bugfix
- [ ] New feature
- [ ] Breaking change

## Testing

- [ ] Does this PR has >90% code coverage?
- [ ] Is the screenshot attached for code coverage?
- [ ] Does the `gradle build` pass in your local? 
- [ ] Is the `gradle build` logs attached?

### CODE REVIEW

File: liquibase-service/src/main/resources/db/rdb/routines/001-generate_data_for_data_sync_config-005.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+DELETE
+FROM [dbo].[data_sync_config]
+WHERE table_name = 'D_INV_TREATMENT';
+
+DELETE
+FROM [dbo].[data_sync_config]
+WHERE table_name = 'D_INV_VACCINATION';
+
+-- Inserting updated query
+INSERT INTO [dbo].[data_sync_config]
```
Comment (upasanapattnaik-eq) [inline liquibase-service/src/main/resources/db/rdb/routines/001-generate_data_for_data_sync_config-005.sql:35]: 👍

**Review by upasanapattnaik-eq — COMMENTED** [review]
The file needs to be added into the rdb changelog.

File: liquibase-service/src/main/resources/db/rdb/routines/001-generate_data_for_data_sync_config-005.sql [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
... F_PAGE_CASE.D_INV_VACCINATION_KEY JOIN INVESTIGATION ON F_PAGE_CASE.INVESTIGATION_KEY = INVESTIGATION.INVESTIGATION_KEY WHERE INVESTIGATION.LAST_CHG_TIME :operator :timestamp;', 'SELECT D_INV_VACCINATION.* FROM D_INV_VACCINATION JOIN F_PAGE_CASE ON D_INV_VACCINATION.D_INV_VACCINATION_KEY = F_PAGE_CASE.D_INV_VACCINATION_KEY JOIN INVESTIGATION ON F_PAGE_CASE.INVESTIGATION_KEY = INVESTIGATION.INVESTIGATION_KEY WHERE INVESTIGATION.LAST_CHG_TIME IS NULL;', 'SELECT COUNT(*) FROM D_INV_VACCINATION JOIN F_PAGE_CASE ON D_INV_VACCINATION.D_INV_VACCINATION_KEY = F_PAGE_CASE.D_INV_VACCINATION_KEY JOIN INVESTIGATION ON F_PAGE_CASE.INVESTIGATION_KEY = INVESTIGATION.INVESTIGATION_KEY WHERE INVESTIGATION.LAST_CHG_TIME :operator :timestamp;', 'WITH PaginatedResults AS (SELECT D_INV_VACCINATION.*, ROW_NUMBER() OVER (ORDER BY INVESTIGATION.LAST_CHG_TIME ASC) AS RowNum FROM D_INV_VACCINATION JOIN F_PAGE_CASE ON D_INV_VACCINATION.D_INV_VACCINATION_KEY = F_PAGE_CASE.D_INV_VACCINATION_KEY JOIN INVESTIGATION ON F_PAGE_CASE.INVESTIGATION_KEY = INVESTIGATION.INVESTIGATION_KEY WHERE INVESTIGATION.LAST_CHG_TIME :operator :timestamp) SELECT * FROM PaginatedResults WHERE RowNum BETWEEN :startRow AND :endRow;')
```
Comment (upasanapattnaik-eq) [inline liquibase-service/src/main/resources/db/rdb/routines/001-generate_data_for_data_sync_config-005.sql:38]: It's a little hard to review the changes as a single block. The default formatting makes it a single line of updates as well. I am not sure if making each insert value block into it's own line is the most feasible solution, but if possible we could try it.

Comment (ndduc01) [reply liquibase-service/src/main/resources/db/rdb/routines/001-generate_data_for_data_sync_config-005.sql]: make the update, let me know how it goes!

File: liquibase-service/src/main/resources/db/rdb/routines/001-generate_data_for_data_sync_config-005.sql [hunk 3]
```diff
@@ -0,0 +1,81 @@
+-- Removing obsoleted query
```
Comment (upasanapattnaik-eq) [inline liquibase-service/src/main/resources/db/rdb/routines/001-generate_data_for_data_sync_config-005.sql:1]: @alog-enquizit Could you please check the changelog for the rdb and confirm if we need to add in the changes? What would the runtime expectation in demo be?

Comment (alog-enquizit) [reply liquibase-service/src/main/resources/db/rdb/routines/001-generate_data_for_data_sync_config-005.sql:1]: @ndduc01 You might want to update the databasechange log to include the scripts for liquibase to pick it up and run them.

https://github.com/CDCgov/NEDSS-DataReporting/blob/main/liquibase-service/src/main/resources/db/changelog/db.rdb.changelog-16.1.yaml

You might have to do something like this:

```changeSet:
      id: 2
      author: liquibase
      changes:
        - sqlFile:
            path: 001-create_data_sync_config-002.sql
            splitStatements: false
```

Comment (ndduc01) [reply liquibase-service/src/main/resources/db/rdb/routines/001-generate_data_for_data_sync_config-005.sql:1]: so I need to do it for 001-create_data_sync_config-003.sql, then 004 and so on as well?

**Review by upasanapattnaik-eq — APPROVED** [review]
LGTM!

File: liquibase-service/src/main/resources/db/changelog/db.rdb.changelog-16.1.yaml [hunk 4]
```diff
@@ -12,4 +12,32 @@ databaseChangeLog:
       changes:
         - sqlFile:
             path: 001-create_data_sync_config-001.sql
+            splitStatements: false
+  - changeSet:
+      id: 2
+      author: liquibase
+      changes:
+        - sqlFile:
+            path: 001-create_data_sync_config-002.sql
```
Comment (alog-enquizit) [inline liquibase-service/src/main/resources/db/changelog/db.rdb.changelog-16.1.yaml:21]: The file path is incorrect. From id:2,  It should be  like this: 001-generate_data_for_data_sync_config-001.sql
id3: path: 001-generate_data_for_data_sync_config-002.sql

Each of the database scripts should be represented in the changelog if we require liquibase to run them.

File: liquibase-service/src/main/resources/db/changelog/db.rdb.changelog-16.1.yaml [hunk 5]
```diff
@@ -12,4 +12,32 @@ databaseChangeLog:
...
+            splitStatements: false
+  - changeSet:
+      id: 3
+      author: liquibase
+      changes:
+        - sqlFile:
+            path: 001-create_data_sync_config-003.sql
```
Comment (alog-enquizit) [inline liquibase-service/src/main/resources/db/changelog/db.rdb.changelog-16.1.yaml:28]: Likewise this should be `001-generate_data_for_data_sync_config-002.sql`

File: liquibase-service/src/main/resources/db/changelog/db.rdb.changelog-16.1.yaml [hunk 6]
```diff
@@ -12,4 +12,32 @@ databaseChangeLog:
...
+            splitStatements: false
+  - changeSet:
+      id: 4
+      author: liquibase
+      changes:
+        - sqlFile:
+            path: 001-create_data_sync_config-004.sql
```
Comment (alog-enquizit) [inline liquibase-service/src/main/resources/db/changelog/db.rdb.changelog-16.1.yaml:35]: `001-generate_data_for_data_sync_config-003.sql`

**Pull request merged**